src/typing/clearTypingNotification.js

```diff
@@ -0,0 +1,31 @@
+/* @flow */
+import type { Dispatch, GetState } from '../types';
+
+import { getTyping } from '../directSelectors';
+import config from '../config';
+
+export const clearTypingNotification = () => async (dispatch: Dispatch, getState: GetState) => {
```

Comment:
Hmm... this looks like quite a convoluted way to implement this.
Would you elaborate why a simpler `setTimeout` call is not enough?

Reply:
I found some difficulties with `timeout`. 
What we want to only dismiss those notifications which are outdated. For example if typing start event arrived at time `t` then it should only be dismissed at time `t+15 sec` if no new start typing event have arrived in between `t` and `t+15 sec`. If new start typing has arrived at `t+5 sec` then we need to take account at `t+5+15 sec` thus invalidating previous `timeout`.

Reply:
I found this to be more simpler than `setTimeout` one. :)